### DISCUSSION TITLE
Combining with laravel-query-builder

### DISCUSSION THREAD

**Discussion by AndreasHerss** [spatie/laravel-data#443] [Q&A]
Hi, i'm trying to combine laravel-data and laravel-query-builder in the same requests, but have an issue, maybe it's just me staring at it too long 🤷‍♂️ 

I'm using laravel-query-builder to allow sort and filters, and the send the data to my DTO made with laravel-data.
The problem occurs when adding an ?include=user,date

The date is just a Lazy field, and the user is a Lazy relation

if not adding the user allowedIncludes to the query, it gives an N+1 issue (as the data is served from the query-builder)
if adding the user i get an error that the date is not allowed. as i'm trying to include the date field too.. 

Any help would be appreciated :)

**Comment by eelco2k** [comment]
Could you provide some more info about the structure of your DTO? and the way you setup the laravel-query-builder... As for now as the includes can be optional i also think the DTO parameters in your constructor should be optional with an `?`.

Just a thought...
